res/cmake/def.cmake

```diff
@@ -2,11 +2,24 @@
 
 # Per compiler CXXFLAGS
 set (PHARE_FLAGS ${PHARE_FLAGS} )
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-  set (PHARE_FLAGS ${PHARE_FLAGS} )
-else() # !Clang
-  set (PHARE_FLAGS ${PHARE_FLAGS} --param=min-pagesize=0 )
-endif() # clang
+
+if(PHARE_COMPILER_WORKAROUNDS)
```

Comment:
this seems complex... how stable is that, do you expect we won't need that soon or it'll have to stay

Reply:
I expect we need the compile flags just cause it is failing without them

this mechanism is just to allow it to be switched off if undesirable for some reason